Server crashes when it receives a message for an unknown entity

Following is the error thrown by the server after it receives a message for the `org.ehcache.clustered.lock.client.VoltronReadWriteLockClient` entity that was not installed on its classpath:

```
com.tc.util.TCAssertionError: Unexpected exception in entity - CRASHING
        at com.tc.util.Assert.failure(Assert.java:41)
        at com.tc.objectserver.handler.ProcessTransactionHandler.addMessage(ProcessTransactionHandler.java:172)
        at com.tc.objectserver.handler.ProcessTransactionHandler.access$200(ProcessTransactionHandler.java:59)
        at com.tc.objectserver.handler.ProcessTransactionHandler$1.handleEvent(ProcessTransactionHandler.java:87)
        at com.tc.objectserver.handler.ProcessTransactionHandler$1.handleEvent(ProcessTransactionHandler.java:70)
        at com.tc.async.impl.StageQueueImpl$HandledContext.runWithHandler(StageQueueImpl.java:502)
        at com.tc.async.impl.StageImpl$WorkerThread.run(StageImpl.java:199)
Caused by: java.lang.IllegalArgumentException: Can't handle entity type org.ehcache.clustered.lock.client.VoltronReadWriteLockClient
        at com.tc.objectserver.entity.ServerEntityFactory.getService(ServerEntityFactory.java:72)
        at com.tc.objectserver.entity.EntityManagerImpl.getVersionCheckedService(EntityManagerImpl.java:156)
        at com.tc.objectserver.entity.EntityManagerImpl.createEntity(EntityManagerImpl.java:99)
        at com.tc.objectserver.handler.EntityExistenceHelpers.createEntityReturnWasCached(EntityExistenceHelpers.java:56)
        at com.tc.objectserver.handler.ProcessTransactionHandler.addMessage(ProcessTransactionHandler.java:150)
        ... 5 more
```

Obviously, the server doesn't know how to handle such entity; but it should not crash but respond to the client with an error that eventually could be processed and transformed into a meaningful exception.

--- a/dso-l2/src/main/java/com/tc/objectserver/entity/ServerEntityFactory.java
+++ b/dso-l2/src/main/java/com/tc/objectserver/entity/ServerEntityFactory.java
@@ -46,17 +46,18 @@ import org.terracotta.entity.ServerEntityService;
 import java.util.List;
 import org.terracotta.entity.EntityMessage;
 import org.terracotta.entity.EntityResponse;
+import org.terracotta.exception.EntityNotFoundException;
 
 /**
  * @author twu
  */
 public class ServerEntityFactory {
-  public static <T extends ServerEntityService<? extends ActiveServerEntity, ? extends PassiveServerEntity>> T getService(String typeName) {
+  public static <T extends ServerEntityService<? extends ActiveServerEntity, ? extends PassiveServerEntity>> T getService(String typeName) throws EntityNotFoundException {
     return getService(typeName, Thread.currentThread().getContextClassLoader());
   }
 
   @SuppressWarnings({ "rawtypes", "unchecked" })
-  public static <T extends ServerEntityService<? extends EntityMessage, ? extends EntityResponse>> T getService(String typeName, ClassLoader classLoader) {
+  public static <T extends ServerEntityService<? extends EntityMessage, ? extends EntityResponse>> T getService(String typeName, ClassLoader classLoader) throws EntityNotFoundException {
     List<Class<? extends ServerEntityService>> serviceLoader = ServiceLocator.getImplementations(ServerEntityService.class, classLoader);
     for (Class<? extends ServerEntityService> serverService : serviceLoader) {
       try {
@@ -69,6 +70,6 @@ public class ServerEntityFactory {
       }
 
     }
-    throw new IllegalArgumentException("Can't handle entity type " + typeName);
+    throw new EntityNotFoundException(typeName, "Can't handle entity type");
   }
 }

